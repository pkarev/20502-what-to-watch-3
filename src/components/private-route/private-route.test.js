import React from 'react';
import renderer from 'react-test-renderer';
import PrivateRoute from './private-route';
import history from '../../history';
import {Router} from 'react-router-dom';
import PropTypes from 'prop-types';

const MockComponent = ({children}) => (
  <div>
    {children}
  </div>
);

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

it(`Render PrivateRoute child component if authorized`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <PrivateRoute isAuthorized={true} path="" render={() => (
            <MockComponent>
              <p>test</p>
            </MockComponent>
          )}
          />
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
