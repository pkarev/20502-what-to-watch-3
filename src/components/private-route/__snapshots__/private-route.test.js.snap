// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PrivateRoute child component if authorized 1`] = `
<div>
  <p>
    test
  </p>
</div>
`;
